onnxruntime_extensions/tools/add_pre_post_processing_to_model.py

```diff
@@ -213,17 +217,19 @@ def yolo_detection(model_file: Path, output_file: Path, output_format: str = 'jp
         need_transpose = output_shape[0] < output_shape[1]
     elif len(model.graph.input) > 1:
         print(""" warning!!
-            The model has more than one inputs and the output shape is not fixed.
-            Output shape would be assumed to be [num_boxes, num_classes+4(reg)].
+            The model possesses more than one inputs, indicating we can't generate the input 
+ data and map to model inputs to execute model by onnx-runtime and we can't get concrete output shape.
+            Output shape will be assumed to be [num_boxes, num_classes+4(reg)].
             If this behavior is not expected, please run the onnx shape inference first to \
-            infer the output shape then run this script again.""")
+            infer the output shape out then run this script again.""")
```

Comment:
I think it's best to keep this simple and remove the `elif` as I think it's adding unnecessary complication and confusion and not a lot of value.

I don't think we need to try and support or detect a model that unexpectedly has multiple required inputs in this predefined handler for yolo. If someone has such a model they can cut-and-paste the implementation and adjust as needed. At the most add an assert in the `else` to check num inputs is 1.

Reply:
Sure.